File: numberGuessingGame.py
import random

def number_guessing_game():
    

    number = random.randint(1, 10) # Generate a number between 1, 10
    tries = 3
    print("Guess a number between 1 and 10!") # Introduction to start the game
    while tries > 0:
        print(f"You have {tries} guesses left.") # Let the player know how many tries they have left
        try:
            guess = int(input("Enter your guess: ")) # Ask the player for their guess
        except ValueError:
            print("Invalid input. Please enter a number.")
            continue

            # Validate the guess in the range and let player know how close
            
        if guess == number:
            print("Correct!")
            continue
        elif guess < number:
            print("Too low!")
        else:
            print("Too high!")

        tries -= 1

    # let player know they are out of guesses

    print("You are out of guesses. The number was", number)
# Start the game when the script is ran
if __name__ == "__main__":
    number_guessing_game()

File: NumberGuessingGame.cpp
#include <iostream>
#include <cstdlib>   // for rand() and srand()
#include <ctime>     // for time()

using namespace std;

int main()
{
    bool done = false;
    int secret;
    int guess;
    int guessCount = 0;

    // Generate a random number between 1 and 100
    srand(time(0));          // seed random number generator
    secret = rand() % 100 + 1;

    cout << "I'm thinking of a number between 1 and 100.\n";

    while (!done)
    {
        cout << "\nGuess the number 1-100! > ";
        cin >> guess;

        guessCount++;

        if (guess == secret)
        {
            cout << "Congrats! You win!\n";
            cout << "You guessed it in " << guessCount << " tries.\n";
            done = true;
        }
        else if (guess < secret)
        {
            cout << "Sorry, your guess is too low.\n";
            cout << "Guesses made: " << guessCount << " / 10\n";
        }
        else
        {
            cout << "Sorry, your guess is too high.\n";
            cout << "Guesses made: " << guessCount << " / 10\n";
        }

        // Check if player exceeded 10 guesses
        if (guessCount >= 10 && !done)
        {
            cout << "\nGame over! You’ve used all 10 guesses.\n";
            cout << "The secret number was: " << secret << endl;
            done = true;
        }
    }

    return 0;
}
